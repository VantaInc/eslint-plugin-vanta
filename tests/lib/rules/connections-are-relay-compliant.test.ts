import { GraphQLRuleTester } from "@graphql-eslint/eslint-plugin";
import rule from "../../../lib/rules/connections-are-relay-compliant";

const ruleTester = new GraphQLRuleTester();

ruleTester.runGraphQLTests("connections-are-relay-compliant", rule, {
  valid: [
    {
      code: `
type Connection {
  edges: [testEdge]
  pageInfo: PageInfo!
}
`,
    },
    {
      code: `
type Connection {
  edges: [testEdge!]!
  pageInfo: PageInfo!
}
`,
    },
    {
      code: `
type Connection {
  edges: [testEdge]!
  pageInfo: PageInfo!
}
`,
    },
    {
      code: `
type Connection {
  edges: [testEdge!]
  pageInfo: PageInfo!
}
`,
    },
  ],
  invalid: [
    {
      code: `
    type MissingEdgesConnection {
      pageInfo: PageInfo!
    }`,
      errors: 1,
    },
    {
      code: `
      type EdgesNotListConnection {
        edges: testEdge
        pageInfo: PageInfo!
      }`,
      errors: 1,
    },
    {
      code: `
      type MissingPageInfoConnection {
        edges: [testEdge]
      }`,
      errors: 1,
    },

    {
      code: `
      type PageInfoNotNonnullableConnection {
        edges: [testEdge]
        pageInfo: PageInfo
      }`,
      errors: 1,
    },
    {
      code: `
      type PageInfoNotObjectTypeConnection {
        edges: [test]
        pageInfo: [test]!
      }`,
      errors: 2,
    },
    {
      code: `
      type EdgeDoesntEndWithEdgeConnection {
        edges: [testEdgeNot!]
        pageInfo: PageInfo!
      }`,
      errors: 1,
    },
    {
      code: `
      type PageInfoWrongTypeConnection {
        edges: [testNotEdge!]
        pageInfo: notPageInfo!
      }
`,
      errors: 1,
    },
  ],
});
