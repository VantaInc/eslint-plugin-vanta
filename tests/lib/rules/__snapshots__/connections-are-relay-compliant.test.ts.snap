// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invalid #1 1`] = `
"#### ⌨️ Code

      1 |     type MissingEdgesConnection {
      2 |       pageInfo: PageInfo!
      3 |     }

#### ❌ Error

    > 1 |     type MissingEdgesConnection {
        |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |       pageInfo: PageInfo!
        | ^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |     }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Connections must contain an edge field"
`;

exports[`Invalid #2 1`] = `
"#### ⌨️ Code

      1 |       type EdgesNotListConnection {
      2 |         edges: testEdge
      3 |         pageInfo: PageInfo!
      4 |       }

#### ❌ Error

    > 1 |       type EdgesNotListConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: testEdge
        | ^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: PageInfo!
        | ^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Edges field must be a list"
`;

exports[`Invalid #3 1`] = `
"#### ⌨️ Code

      1 |       type MissingPageInfoConnection {
      2 |         edges: [testEdge]
      3 |       }

#### ❌ Error

    > 1 |       type MissingPageInfoConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [testEdge]
        | ^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Connections must contain a pageInfo field"
`;

exports[`Invalid #4 1`] = `
"#### ⌨️ Code

      1 |       type PageInfoNotNonnullableConnection {
      2 |         edges: [testEdge]
      3 |         pageInfo: PageInfo
      4 |       }

#### ❌ Error

    > 1 |       type PageInfoNotNonnullableConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [testEdge]
        | ^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: PageInfo
        | ^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pageInfo must be non-nullable"
`;

exports[`Invalid #5 1`] = `
"#### ⌨️ Code

      1 |       type PageInfoNotObjectTypeConnection {
      2 |         edges: [test]
      3 |         pageInfo: [test]!
      4 |       }

#### ❌ Error 1/2

    > 1 |       type PageInfoNotObjectTypeConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [test]
        | ^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: [test]!
        | ^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ edge type must have a suffix \`Edge\`

#### ❌ Error 2/2

    > 1 |       type PageInfoNotObjectTypeConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [test]
        | ^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: [test]!
        | ^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pageInfo must be of type PageInfo"
`;

exports[`Invalid #6 1`] = `
"#### ⌨️ Code

      1 |       type EdgeDoesntEndWithEdgeConnection {
      2 |         edges: [testEdgeNot!]
      3 |         pageInfo: PageInfo!
      4 |       }

#### ❌ Error

    > 1 |       type EdgeDoesntEndWithEdgeConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [testEdgeNot!]
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: PageInfo!
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ edge type must have a suffix \`Edge\`"
`;

exports[`Invalid #7 1`] = `
"#### ⌨️ Code

      1 |       type PageInfoWrongTypeConnection {
      2 |         edges: [testNotEdge!]
      3 |         pageInfo: notPageInfo!
      4 |       }

#### ❌ Error

    > 1 |       type PageInfoWrongTypeConnection {
        |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 2 |         edges: [testNotEdge!]
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 3 |         pageInfo: notPageInfo!
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    > 4 |       }
        | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ pageInfo must be of type PageInfo"
`;
